### PR TITLE
Add load-interval properties to interface

### PR DESCRIPTION
**Pull request by saichint**
This PR is for adding load-interval delay properties to interface provider. All tests pass on all platforms.

### CODE REVIEW

File: lib/cisco_node_utils/interface.rb [hunk 1]
```diff
@@ -806,6 +806,61 @@ def feature_lacp_set(val)
       config_set('interface', 'feature_lacp', state: val ? '' : 'no')
     end
 
+    def load_interval_counter_1_delay
+      return nil if @name[/loop/i]
+      config_get('interface', 'load_interval_counter_1_delay', name: @name)
+    end
+
+    def load_interval_counter_1_delay=(val)
+      fail ArgumentError, 'Interface cannot be loopback' if @name[/loop/i]
```
Comment (mikewiebe) [inline lib/cisco_node_utils/interface.rb]: Are there any other logical interfaces we should consider or only loopbacks?

File: lib/cisco_node_utils/interface.rb [hunk 2]
```diff
@@ -806,6 +806,61 @@ def feature_lacp_set(val)
       config_set('interface', 'feature_lacp', state: val ? '' : 'no')
     end
 
+    def load_interval_counter_1_delay
+      return nil if @name[/loop/i]
```
Comment (mikewiebe) [inline lib/cisco_node_utils/interface.rb]: Check to see if this `return nil` is even needed for the `puppet resource` command.

**Comment by saichint** [comment]
Done with changes

File: lib/cisco_node_utils/interface.rb [hunk 3]
```diff
@@ -806,6 +806,62 @@ def feature_lacp_set(val)
...
+      if @name[/(vlan|bdi)/i]
+        config_get_default('interface',
+                           'load_interval_counter_1_delay_vlan_bdi')
+      else
+        config_get_default('interface', 'load_interval_counter_1_delay')
+      end
+    end
+
+    def load_interval_counter_2_delay
+      return nil if @name[/loop/] || @name[/ethernet.*\S+\.\d+$/]
```
Comment (mikewiebe) [inline lib/cisco_node_utils/interface.rb:832]: Can you add a test for this nil condition for both loopbacks and subinterfaces?

File: lib/cisco_node_utils/interface.rb [hunk 4]
```diff
@@ -806,6 +806,62 @@ def feature_lacp_set(val)
...
+      config_get('interface', 'load_interval_counter_2_delay', name: @name)
+    end
+
+    def load_interval_counter_2_delay=(val)
+      fail ArgumentError, 'Interface cannot be sub-intf or loopback' if
```
Comment (mikewiebe) [inline lib/cisco_node_utils/interface.rb:837]: Add a test for subinterfaces and loopbacks to verify this ArgumentError.  You can probably just loop through all 3 getters and setters.

**Comment by mikewiebe** [comment]
@saichint Couple more comments to add negative test cases.

File: lib/cisco_node_utils/interface.rb (hunk 3 above)
Comment (saichint) [reply lib/cisco_node_utils/interface.rb:832]: added

File: lib/cisco_node_utils/interface.rb (hunk 4 above)
Comment (saichint) [reply lib/cisco_node_utils/interface.rb:837]: added

Comment (saichint) [reply lib/cisco_node_utils/interface.rb:837]: I did not add loop for get/set. I think the tests should be clean and readable. If you disagree, I will change it.

**Comment by saichint** [comment]
let me know if you have more comments
